src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1260,7 +1236,19 @@ void MarkEntireAssembly (AssemblyDefinition assembly)
 			MarkCustomAttributes (assembly.MainModule, new DependencyInfo (DependencyKind.AssemblyOrModuleAttribute, assembly.MainModule), null);
 
 			if (assembly.MainModule.HasExportedTypes) {
-				// TODO: This needs more work accross all steps
+				foreach (var exportedType in assembly.MainModule.ExportedTypes) {
+					_context.Annotations.Mark (exportedType, new DependencyInfo (DependencyKind.ExportedType, exportedType));
```

Comment:
This looks like a significant behavior change:
- We didn't used to keep type forwards in copy assemblies - this could result in some extra facade dlls being kept that weren't needed previously. It's probably uncommon, since facades are usually going to be copyused as part of the framework, but it's unclear how copyused is supposed to behave now. Should a marked copyused assembly also have exported types kept? (This would be a bigger issue since copyused are more likely to pull in facades).
- I *think* this will prevent SweepStep from sweeping assembly refs from copy assemblies, so it will never update scopes for Copy assemblies. Same question - how should this work for copyused assemblies?

If we go this direction and mark all copy type forwarders, I think we can simplify a few cases in SweepStep:
- get rid of the copy case from `UpdateAssembliesReferencingRemovedAssembly`
- get rid of the `SweepTypeForwarders` handling for copy assemblies